Tests/TuistCacheTests/Extensions/GraphTraverser+ExtrasTests.swift

```diff
@@ -0,0 +1,157 @@
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+import XCTest
+@testable import TuistCache
+
+class GraphTraverserExtrasTests: XCTestCase {
+    func test_filterIncludedTargets_includeExcluded() throws {
+        // Given
+        let app = Target.test(name: "App", product: .app)
+        let tests1 = Target.test(name: "AppTests1", product: .unitTests)
+        let tests2 = Target.test(name: "AppTests2", product: .unitTests)
+        let tests3 = Target.test(name: "AppTests3", product: .unitTests)
+        let project = Project.test(
+            path: "/path/a"
+        )
+
+        // Given: Value Graph
+        let testDependencies: Set<GraphDependency> = [
+            .target(name: app.name, path: project.path)
+        ]
+        let dependencies: [GraphDependency: Set<GraphDependency>] = [
+            .target(name: app.name, path: project.path): [],
+            .target(name: tests1.name, path: project.path): testDependencies,
+            .target(name: tests2.name, path: project.path): testDependencies,
+            .target(name: tests3.name, path: project.path): testDependencies,
+        ]
+        let graph = Graph.test(
+            projects: [project.path: project],
+            targets: [project.path: [
+                app.name: app,
+                tests1.name: tests1,
+                tests2.name: tests2,
+                tests3.name: tests3,
+            ]],
+            dependencies: dependencies
+        )
+        let subject = GraphTraverser(graph: graph)
+
+        var filteredTargets: Set<GraphTarget>
+        // When
+        filteredTargets = subject.filterIncludedTargets(
+            basedOn: subject.allTargets(),
+            testPlan: nil,
+            includedTargets: [tests1.name],
+            excludedTargets: []
+        )
+
+        // Then
+        XCTAssertEqual(filteredTargets.map(\.target), [tests1])
+
+        // When
+        filteredTargets = subject.filterIncludedTargets(
+            basedOn: subject.allTargets(),
+            testPlan: nil,
+            includedTargets: [tests1.name],
+            excludedTargets: [tests1.name]
+        )
```

Comment:
Works for me! My intent initially was to avoid duplicating the inputs of tests too much. I'll see if I can replace them with properties and not make it too confusing.